### PR TITLE
Overlapping intervals check correction

### PR DESCRIPTION
**Pull request by eddie-c-davis**
## Description

This PR corrects an issue with horizontal regions where multiple computation blocks have the same vertical interval but different horizontal intervals that results in the overlapping intervals exception being thrown when parsing the frontend code. The correction is to check whether overlapping vertical intervals are NOT equal AND overlapping.

## Requirements

Before submitting this PR, please make sure:

- [x] The code builds cleanly without new errors or warnings
- [x] The code passes all the existing tests
- [x] If this PR adds a new feature, new tests have been added to test these
new features

### CODE REVIEW

File: src/gt4py/ir/nodes.py [hunk 1]
```diff
@@ -691,6 +691,14 @@ class AxisInterval(Node):
     end = attribute(of=AxisBound)
     loc = attribute(of=Location, optional=True)
 
+    def equals(self, other: "AxisInterval") -> bool:
```
Comment (jdahm) [inline src/gt4py/ir/nodes.py:694]: Consider implementing `__eq__` instead of making an `equals` method.

Comment (eddie-c-davis) [reply src/gt4py/ir/nodes.py:694]: I did try that first, but did not work so implemented this just get a fix posted. Can try `__eq__` again.

Comment (tehrengruber) [reply src/gt4py/ir/nodes.py]: Overriding `__eq__` is not possible with `attribclass`es. To exclude `loc` from the equality check one needs to add `eq=False` to the attribute, i.e.
```
loc = attribute(of=Location, optional=True, eq=False)
```
To maintain consistency of the semantics of `__eq__` with the other nodes we would also need to do this for all `loc` attributes elsewhere, which I would avoid in the dated frontend. I would propose to use a method named `is_same_interval` instead.

File: src/gt4py/ir/nodes.py [hunk 2]
```diff
@@ -691,6 +691,17 @@ class AxisInterval(Node):
     end = attribute(of=AxisBound)
     loc = attribute(of=Location, optional=True)
 
+    def __eq__(self, other: "AxisInterval") -> bool:
```
Comment (tehrengruber) [inline src/gt4py/ir/nodes.py:694]: As mentioned in my comment this is not working.

File: src/gt4py/frontend/gtscript_frontend.py [hunk 3]
```diff
@@ -862,7 +862,9 @@ def _visit_with_horizontal(
     def _are_intervals_nonoverlapping(self, compute_blocks: List[gt_ir.ComputationBlock]):
         for i, block in enumerate(compute_blocks[1:]):
             other = compute_blocks[i]
-            if not block.interval.disjoint_from(other.interval):
+            if block.interval != other.interval and not block.interval.disjoint_from(
```
Comment (tehrengruber) [inline src/gt4py/frontend/gtscript_frontend.py:865]: I see the point, but we shouldn't do this as the following should still be illegal:
```python
with computation(PARALLEL):
    with interval(0, 3):
        field = 0
    with interval(0, 3):
        field = 1
```
Instead we should check if either the intervals are disjoint or the horizontal regions are disjoint.

**Review by tehrengruber — CHANGES_REQUESTED** [review]

**Comment by eddie-c-davis** [comment]
[PR #483](https://github.com/GridTools/gt4py/pull/483) resolves this issue, so closing.

**Pull request closed without merging**